pennylane/templates/repeat.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``repeat`` template constructor.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+
+###################
+
+
+@template
+def repeat(circuit, depth, *args, **kwargs):
+    r"""Repeatedly applies a function containing quantum gates or templates.
+
+    Args:
+        circuit (function): A function that applies the quantum gates/templates being repeated.
+        depth (int): The number of time ``circuit`` is repeatedly applied.
```

Comment:
```suggestion
        depth (int): The number of times ``circuit`` is repeatedly applied.
```
+1 cent. Also, why are we using ``circuit`` instead of just circuit?